File: AccountTransferMvpApp/AccountTransferMvpAppTests/Utils.Tests.Unit/ServiceLocatorTests.m
#import "ServiceLocatorTests.h"
#import "AccountTransferPresenter.h"
#import "PresenterTypeEnum.h"
#import "ServiceLocator.h"

@implementation ServiceLocatorTests

-(void)testGivenSut_whenResolveIsPassedEnumValue_thenMatchingPresenterIsReturned {
    PresenterTypeEnum presenterTypeEnum = AccountTransfer;
    id accountTransferPresenter = [ServiceLocator resolve:presenterTypeEnum];

    STAssertTrue([accountTransferPresenter isKindOfClass:[AccountTransferPresenter class]], @"The expected type should have been returned.");
}

@end

File: AccountTransferMvpPApp/UnitTests/src/com/gaddzeit/utils/tests/unit/ServiceLocatorTests.java
package com.gaddzeit.utils.tests.unit;

import com.gaddzeit.presenters.AccountTransferPresenter;
import com.gaddzeit.presenters.Presenterable;
import com.gaddzeit.utils.PresenterTypeEnum;
import com.gaddzeit.utils.ServiceLocator;
import org.junit.Test;

import static org.junit.Assert.assertTrue;

public class ServiceLocatorTests {
    @Test
    public void givenSut_whenResolveMethodPassedAnEnumValue_thenMatchingPresenterReturned() {
        PresenterTypeEnum presenterTypeEnum = PresenterTypeEnum.AccountTransfer;
        Presenterable accountTransferPresenter = ServiceLocator.resolve(presenterTypeEnum);

        assertTrue("Returned type should match", accountTransferPresenter instanceof AccountTransferPresenter);
    }
}
